backbone.js

```diff
@@ -1088,8 +1088,8 @@
     },
 
     // Define how to uniquely identify models in the collection.
-    modelId: function(attrs) {
-      return attrs[this.model.prototype.idAttribute || 'id'];
+    modelId: function(attrs, model) {
+      return attrs[(model && model.idAttribute) || this.model.prototype.idAttribute || 'id'];
```

Comment:
Yeah, Backbone as it is assumes polymorphic collections won't contain models of different types with the same id: http://jsfiddle.net/53oo2bo9/ 

That doesn't feel intuitive to me, but as it is now, this PR will still work fine :)

I like the idea of just passing the idAttribute, will update the PR